File: v2/core/lib/repository/i_clipboard_repository.dart
import '../models/card_color.dart';
import '../models/clipboard_content_type.dart';
import '../models/clipboard_item.dart';

abstract interface class IClipboardRepository {
  Future<void> save(ClipboardItem item);
  Future<void> update(ClipboardItem item);
  Future<ClipboardItem?> getById(String id);
  Future<ClipboardItem?> getLatest();
  Future<ClipboardItem?> findByContentAndType(
    String content,
    ClipboardContentType type,
  );
  Future<ClipboardItem?> findByContentHash(String contentHash);
  Future<List<ClipboardItem>> getAll();
  Future<void> delete(String id);
  Future<int> clearOldItems(int days, {bool excludePinned = true});
  Future<List<ClipboardItem>> search(
    String query, {
    int limit = 50,
    int skip = 0,
  });
  Future<List<ClipboardItem>> searchAdvanced({
    String? query,
    List<ClipboardContentType>? types,
    List<CardColor>? colors,
    bool? isPinned,
    required int limit,
    required int skip,
  });
  Future<void> close();
}

File: v1/CopyPaste.Core/IClipboardRepository.cs
namespace CopyPaste.Core;

public interface IClipboardRepository
{
    void Save(ClipboardItem item);
    void Update(ClipboardItem item);
    ClipboardItem? GetById(Guid id);
    ClipboardItem? GetLatest();
    ClipboardItem? FindByContentAndType(string content, ClipboardContentType type);
    IEnumerable<ClipboardItem> GetAll();
    void Delete(Guid id);
    int ClearOldItems(int days, bool excludePinned = true);
    IEnumerable<ClipboardItem> Search(string query, int limit = 50, int skip = 0);

    /// <summary>
    /// Advanced search with all filters applied at DB level.
    /// When any filter is active, searches ALL items (ignores isPinned tab filter).
    /// </summary>
    IEnumerable<ClipboardItem> SearchAdvanced(
        string? query,
        IReadOnlyCollection<ClipboardContentType>? types,
        IReadOnlyCollection<CardColor>? colors,
        bool? isPinned,
        int limit,
        int skip);

    /// <summary>
    /// Finds an item by its ContentHash (indexed for fast lookups).
    /// Primarily used for image deduplication.
    /// </summary>
    ClipboardItem? FindByContentHash(string contentHash);
}
